### PR TITLE
Fix image building UI

### PR DESCRIPTION
**Pull request by Etheryte**
## What does this PR change?

Adds optional chaining for image building UI, fixing https://bugzilla.suse.com/show_bug.cgi?id=1186728  
Also fixes an unrelated lifecycle bug I found on the way.

## GUI diff

No difference, except now you actually see the UI.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: only internal and user invisible changes

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/15076 and https://bugzilla.suse.com/show_bug.cgi?id=1186728

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: web/html/src/build/fill-spec-file.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ function fillSpecFile() {
         }
         return item;
     });
-    const mappedProcessedLicenses = Array.from(new Set(processedLicenses)).sort().join(" and ");
+    const mappedProcessedLicenses = Array.from(new Set(processedLicenses)).sort().join(" AND ");
```
Comment (cbbayburt) [inline web/html/src/build/fill-spec-file.js:13]: Is this change also intended?

**Review by cbbayburt — APPROVED** [review]
I haven't tested it manually, but the code looks good to me :+1: Thanks for the quick fix!

File: web/html/src/build/fill-spec-file.js (hunk 1 above)
Comment (Etheryte) [reply web/html/src/build/fill-spec-file.js:13]: Yes, this is for [SUSE/spacewalk#15002](https://github.com/SUSE/spacewalk/issues/15002), I either fix it here or mess up the licenses in master if this gets merged.

Comment (cbbayburt) [reply web/html/src/build/fill-spec-file.js:13]: ok, all good then :+1:

**Review by parlt91 — APPROVED** [review]
LGTM Thanks

**Pull request merged**